src/server/cluster_family_test.cc

```diff
@@ -458,6 +458,11 @@ TEST_F(ClusterFamilyTest, ClusterConfigDeleteSlots) {
   EXPECT_THAT(slot2, ElementsAre("2", "key_count", "0"));
 }
 
+TEST_F(ClusterFamilyTest, ClusterModeSelectNotAllowed) {
+  EXPECT_THAT(Run({"select", "1"}).GetString(), "ERR SELECT is not allowed in cluster mode");
```

Comment:
It's better to use `ErrArg()` instead of directly comparing to string, as that makes sure that an actual error is sent over the line (i.e. starts with `-`)
And while at it:
1. The `GetString()` parts here is unneeded in these tests, as the test library provides an `operator==` with `string_view`s
2. Style-wise, I think it's clearer to write `EXPECT_EQ()` instead of `EXPECT_THAT()` when doing equality checks

Reply:
Thanks I fixed.
BTW the ErrArg does not check that the error starts with - as not all errors start with -.
Not sure what is the convetion, when they should start with - and when not.
The Select error does not start with -

Reply:
To my understanding, in Redis protocol all errors start with a `-`.
This is not true for memcached (see [here](https://shaharmike.com/dfly/dragonfly/src/facade/reply_builder.cc.html#150) - we just send strings), but for our Redis reply builder, when we call `SendError()`, it checks if the strings starts with a `-`. If it does, it sends the strings as is, if it doesn't, then it prefixes it with `-ERR `, see [here](https://shaharmike.com/dfly/dragonfly/src/facade/reply_builder.cc.html#196)).
So I guess that if we're explicit about an error code we should call `SendError("-CODE ...")`, otherwise we can use a general error.
See also the RESP specifications: https://redis.io/docs/reference/protocol-spec/#resp-errors